### DISCUSSION TITLE
add `useReducerAtom` with `selector` option?

### DISCUSSION THREAD

**Discussion by himself65** [pmndrs/jotai#2289] [Ideas] [answered]
I'm refactoring the old project that uses redux. I hope there's something like the selector in Zustand

```
const [xxx, dispatch] = useReducerAtom(appState, appReducer, state => state.xxx)
```

**Comment by dai-shi** [comment]
To my knowledge, reducer atom utils are not widely used. They are introduced in early days, and exist for historical reasons. Basically, instead of adding utils, the idea is to have such recipes in docs nowadays.

I think you can implement (I can help if you want) your own `useReducerAtom`, but one caveat is you would need a stable selector with useCallback or useRef.

**Reply by himself65** [reply]
zustand use `useSyncExternalStoreWithSelector` to reduce the re-render. I'm not sure if I only use `useRef` `useCallback` with `useAtom` could do the same thing?

**Reply by dai-shi** [reply, marked as answer]
The basic idea is this:
```js
const useAtomSelector = (baseAtom, selector) => {
  const memoizedAtom = useMemo(() => atom((get) => selector(get(baseAtom))), [baseAtom, selector]);
  return useAtomValue(memoizedAtom);
};
```
